app/services/translation_service/store/translation.rb

```diff
@@ -0,0 +1,201 @@
+module TranslationService::Store::Translation
+
+  CommunityTranslationModel = ::CommunityTranslation
+
+  Translation = EntityUtils.define_builder(
+    [:translation_key, :mandatory, :string],
+    [:locale, :mandatory, :string],
+    [:translation])
+
+
+  module_function
+
+  # Create translations
+  # Format for translation_groups:
+  # [ { translation_key: nil // optional key - if defined it will override previous translations
+  #   , translations:
+  #     [ { locale: "en-US"
+  #       , translation: "Welcome"
+  #       }
+  #     , { locale: "fi-FI"
+  #       , translation: "Tervetuloa"
+  #       }
+  #     ]
+  #   }
+  # ]
+  def create(community_id:, translation_groups: [])
+    if translation_groups.empty?
+      msg = "You must specify 'translation_groups' as an array of hash-objects containing translation_key and array of translations - like translation_groups: [ { translation_key: nil, translations: [ { locale: 'en-US' , translation: 'Hi!'}, { locale: 'fi-FI', translation: 'Moi!'}]}]"
+      raise ArgumentError.new(msg)
+    end
```

Comment:
Oh ok, now I see your point. I think you go a long with avoiding code dup with private helper functions. And then when you need it, you have more context, like api method name, etc., at API endpoint to produce an error msg to user than in some later shared part of the code path. A bit generic comment but as a principle I like API validates what comes in and further down the road we can trust more on input data.